proofs/tree.go

```diff
@@ -185,6 +194,9 @@ func ValueToString(value interface{}) (s string, err error) {
 		return strconv.FormatInt(value.(int64), 10), nil
 	case reflect.TypeOf([]uint8{}):
 		return base64.StdEncoding.EncodeToString(value.([]uint8)), nil
+	case reflect.TypeOf(&timestamp.Timestamp{}):
+		v := reflect.ValueOf(value).Elem().Interface().(timestamp.Timestamp)
```

Comment:
I believe `reflect.ValueOf(value)` already returns `Value` type. I do not think `Elem()` is needed.

Reply:
the value is actually of type *timestamp.Timestmap (i.e. it is a pointer, therefore I need to dereference it with Elem()

Reply:
Ah ok. I didn't realized that TimestampProto returns a pointer. Misleading as the other supported types are passed by value. On the other hand, timestamp is the first that is a more "complex" struct, and usually structs are passed by reference. All good.

Reply:
Although, thinking about it a bit more, as we are not in control on how the timestamp is defined in the complex struct to be parsed, I would add another case where we support the dereferenced value as well. Like:
```
case reflect.TypeOf(timestamp.Timestamp{}):
  v := reflect.ValueOf(value).Interface().(timestamp.Timestamp)
```
Or even better, enforcing dereferencing when passing the value to this function.